File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
void calc_coleman_liau(int letters, int words, int sentences);

int main(void)
{
    string text = get_string("Text: ");
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);
    calc_coleman_liau(letters, words, sentences);
}

int count_letters(string text)
{
    int letters = 0;
    for (int i = 0, length = strlen(text); i < length; i++)
    {
        // check if char is a letter
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    int words = 0;
    for (int i = 0, length = strlen(text); i < length; i++)
    {
        //  check if char is white space
        if (isblank(text[i]))
        {
            words++;
        }
    }
    //  adds the last word to the count
    words++;

    return words;
}

int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0, length = strlen(text); i < length; i++)
    {
        // check if char is end of sentence punctuation
        if ((text[i] == '.') || (text[i] == '!') || (text[i] == '?'))
        {
            sentences++;
        }
    }
    return sentences;
}

void calc_coleman_liau(int letters, int words, int sentences)
{
    float l = ((float) letters / words) * 100;
    float s = ((float) sentences / words) * 100;
    double index = round(0.0588 * l - 0.296 * s - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.0f\n", index);
    }
}

File: sentimental-readability/readability.py
from cs50 import get_string


def main():
    txt = get_string("Text: ")

    letters = 0
    words = 1  # Start at 1 because the loop does not account for the last word.
    sentences = 0
    for c in txt:
        if c.isalpha():
            letters += 1
        elif c.isspace():
            words += 1
        elif c in [".", "!", "?"]:
            sentences += 1

    # Average letters and sentences per 100 words
    L = (letters / words) * 100
    S = (sentences / words) * 100

    # Coleman-Liau index Formula
    index = int(round(0.0588 * L - 0.296 * S - 15.8))

    # Print result index as grade
    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


if __name__ == "__main__":
    main()
